src/seedu/addressbook/data/person/Address.java

```diff
@@ -34,22 +48,27 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
-
+    
+	public String getValue() {
+		return block.getBlock() + ", " + street.getStreet() + ", " 
+        	   + unit.getUnit() + ", " + postalCode.getPostalCode();
+	}
+    
     @Override
     public String toString() {
-        return value;
+        return this.getValue();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
```

Comment:
you can also implement equals method in each class and use that here. ( of course that would be more work :-P )